### ISSUE TITLE
What are dimensions?

### ISSUE THREAD

**Issue by RomanSteinberg** [supabase/vecs#98]
**Describe the bug**
I have standard document table in Supabase:
```sql
create table documents (
  id bigserial primary key,
  content text, -- corresponds to Document.pageContent
  metadata jsonb, -- corresponds to Document.metadata
  embedding vector(1536) -- 1536 works for OpenAI embeddings, change if needed
);
```
My vecs code:
```python
client = vecs.create_client(connection_uri)
collection = client.get_or_create_collection(name='documents', dimension=1536)
```
This code raises an MismatchedDimension in Collection._create_if_not_exists function. It says that 3 is not equal to 1536.
This function executes query
```sql
select
    relname as table_name,
    atttypmod as embedding_dim
from
    pg_class pc
    join pg_attribute pa
        on pc.oid = pa.attrelid
where
    pc.relnamespace = 'vecs'::regnamespace
    and pc.relkind = 'r'
    and pa.attname = 'vec'
    and not pc.relname ^@ '_'
    and pc.relname = 'documents'
```
and it returns 3. But if you look into my table there is no number 3 at all. If one wishes to get size of vector should use
```sql
SELECT vector_dims(embedding) AS embedding_dim
FROM documents
LIMIT 1;
```
May be I do not understand what is dimension for vecs collection. So, please, explain.

**To Reproduce**
1. Install vecs 0.4.4
2. Create table in Supabase mentioned above.
3. Run client.get_or_create_collection (code above)

**Expected behavior**
I expect correct obtaining of the collection.

**Screenshots**

**Versions:**
 - PostgreSQL: 15.6 [supabase.com]
 - vecs version:  0.4.4

**Comment by olirice** [comment]
you are correct that the dimension should be 1536 in this case but `vecs` creates the table for you, you should not be writing any SQL directly.

I'd suggest trying a different collection name that doesn't already exist and seeing if that resolves the issue

**Comment by RomanSteinberg** [comment]
> you are correct that the dimension should be 1536 in this case but `vecs` creates the table for you, you should not be writing any SQL directly.
> 
> I'd suggest trying a different collection name that doesn't already exist and seeing if that resolves the issue

@olirice ook into my code sample please. I choose different collection `documents` but library ignores it. 

I investigated little bit more. Library always query schema `vecs`. [Reference](https://github.com/supabase/vecs/blob/main/src/vecs/collection.py#L223).

**Comment by olirice** [comment]
> Library always query schema vecs

correct, the `vecs` library creates and manages all tables in the `vecs` schema of the postgres instance. That is intended behavior

> look into my code sample please

the issue is that you manually ran the code
```sql
create table documents (
  id bigserial primary key,
  content text, -- corresponds to Document.pageContent
  metadata jsonb, -- corresponds to Document.metadata
  embedding vector(1536) -- 1536 works for OpenAI embeddings, change if needed
);
```
you can skip that step entirely. `vecs` will create the table that it needs automatically


If this 
```
select
    relname as table_name,
    atttypmod as embedding_dim
from
    pg_class pc
    join pg_attribute pa
        on pc.oid = pa.attrelid
where
    pc.relnamespace = 'vecs'::regnamespace
    and pc.relkind = 'r'
    and pa.attname = 'vec'
    and not pc.relname ^@ '_'
    and pc.relname = 'documents'
```
is returning `3` then it means your table was inadvertently created with
```
embedding vector(3)
```

I suspect what has happened is that you 
- manually created the documents table
- ran `collection = client.get_or_create_collection(name='documents', dimension=3)` once
-   - which created an entirely new table in the `vecs` schema named `vecs.documents`
- updated the python call to `collection = client.get_or_create_collection(name='documents', dimension=1536)`
-   - and that call now fails because there is an existing table named `vecs.documents` with vector(3) as the dimenison


To resolve that, go into your supabase project (note, this will delete any existing vecs data) and run
```
drop schema vecs cascade;
```

Once you've done that, you can start over with
```
client = vecs.create_client(connection_uri)
collection = client.get_or_create_collection(name='documents', dimension=1536)
```

**Comment by RomanSteinberg** [comment]
@olirice I can see that it is expected behavior. Ok, but I have filled DB at this moment. And it was created according to Supabase manuals with the `documents` table as I presented in `public` schema. Also, I expect that there would be relations with other tables in public schema. So, it seems that `vecs` library is not my choice. Am I right?

PS: Actually, I used `supabase` library to make all queries  I need couple of days ago. So, I have the workaround already.

**Comment by Glinte** [comment]
I think it is quite a reasonable expectation that `vecs` would be able to handle vectors on the public schema too, I guess I have to just adapt the code into my repo

**Comment by olirice** [comment]
> Supabase manuals with the documents table as I presented in public schema. Also, I expect that there would be relations with other tables in public schema in the future. So, it seems that vecs library is not my choice. Am I right?

Correct, `vecs` would not be the right library choice for you if you need more advanced SQL features, full control over the table, content in the public schema, or foreign key references. It makes an assumption that you're primarily interested in the python layer and may be more appropriate for data science type workloads. If you're building a production application with embeddings as a small part of a larger workflow you'd be best served by writing SQL and using the supabase-py client
